platform/external-system-impl/src/com/intellij/openapi/externalSystem/service/project/manage/SdkDataService.kt

```diff
@@ -39,7 +39,7 @@ internal class ProjectSdkDataService : AbstractProjectDataService<ProjectSdkData
     val sdk = projectJdkTable.findJdk(sdkName)
     val projectRootManager = ProjectRootManager.getInstance(project)
     val projectSdk = projectRootManager.projectSdk
-    if (projectSdk == null) {
+    if (projectSdk != sdk) {
```

Comment:
We cannot identify whether Project SDK was defined automatically or whether it was changed in purpose or manually.
We have an issue with that: https://youtrack.jetbrains.com/issue/IDEA-180986

Could you revert this small change?